src/raft.rs

```diff
@@ -726,19 +726,26 @@ impl<T: Storage> Raft<T> {
     /// Appends a slice of entries to the log. The entries are updated to match
     /// the current index and term.
     pub fn append_entry(&mut self, es: &mut [Entry]) {
-        let mut li = self.raft_log.last_index();
+        let li = self.raft_log.last_index();
         for (i, e) in es.iter_mut().enumerate() {
             e.term = self.term;
             e.index = li + 1 + i as u64;
         }
-        // use latest "last" index after truncate/append
-        li = self.raft_log.append(es);
+        self.raft_log.append(es);
 
-        let self_id = self.id;
-        self.mut_prs().get_mut(self_id).unwrap().maybe_update(li);
+        // Not move self's pr.matched until self.on_sync
```

Comment:
here only not update self's pr.matched, but seems in `handle_append_response` it still will update matched for other peers?

Reply:
Yes, will be as you said, I think it's supposed to be that